src/BenchmarkDotNet.Core/Helpers/ProcCpuInfoParser.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using JetBrains.Annotations;
+
+namespace BenchmarkDotNet.Helpers
+{
+    public class ProcCpuInfoParser
+    {
+        public ProcCpuInfoParser(string content)
+        {
+            var logicalCores = Parse(content, ':');
+            var processorsToPhysicalCoresCount = new Dictionary<string, int>();
+            var processorModelNames = new HashSet<string>();
+            foreach (var logicalCore in logicalCores)
+            {
+                if (logicalCore.TryGetValue(ProcCpuInfoKeyNames.PhysicalId, out string physicalId) &&
+                    logicalCore.TryGetValue(ProcCpuInfoKeyNames.CpuCores, out string cpuCoresValue) &&
+                    int.TryParse(cpuCoresValue, out int cpuCoresCount))
+                    processorsToPhysicalCoresCount[physicalId] = cpuCoresCount;
+
+                if (logicalCore.TryGetValue(ProcCpuInfoKeyNames.ModelName, out string modelName))
+                    processorModelNames.Add(modelName);
+            }
+
+            PhysicalProcessorsCount = processorsToPhysicalCoresCount.Count == 0 ? (int?) null : processorsToPhysicalCoresCount.Count;
+            PhysicalCoresCount = processorsToPhysicalCoresCount.Count == 0 ? (int?) null : processorsToPhysicalCoresCount.Values.Sum();
+            ProcessorName = processorModelNames.Count == 0 ? null : string.Join(", ", processorModelNames);
+        }
+
+        public int? PhysicalCoresCount { get; }
+
+        public int? PhysicalProcessorsCount { get; }
+
+        public string ProcessorName { get; }
+
+        [NotNull]
+        private static List<Dictionary<string, string>> Parse([CanBeNull] string content, char separator)
+        {
+            var items = new List<Dictionary<string, string>>();
+            Dictionary<string, string> units = null;
+
+            var list = Regex.Split(content ?? "", "\r?\n");
+            foreach (string line in list)
+                if (line.IndexOf(separator) != -1)
+                {
+                    var lineParts = line.Split(separator);
+                    if (lineParts.Length >= 2)
+                    {
+                        if (units == null)
+                            items.Add(units = new Dictionary<string, string>());
```

Comment:
I don't like tricks like `items.Add(units = new Dictionary<string, string>())`. It looks nice and smart, but it usually leads to bugs when somebody else tries to modify such code. Also, this method looks complicated. Could we simplify it? For example, we can introduce `[CanBeNull] Dictionary<string, string> ParseSection` method, it should make this code more readable.